### PR TITLE
[Improvement]: Honor "Cache-Control: no-store" response header in ful…

### LINKED ISSUE #14429: [Improvement]: Should FullPageCacheListener::responseCanBeCached() honor header "Cache-Control: no-store"?
### Improvement description

As per [RFC 7234](https://www.rfc-editor.org/rfc/rfc7234#section-5.2.1.5)

> The "no-store" request directive indicates that a cache MUST NOT store any part of either this request or any response to it. This directive applies to both private and shared caches.

To me it makes sense that this also applies to the Pimcore Full Page cache.

I know that there are options to disable the full page cache in your application code (either via a `FullPageCacheListener::disable()` call or by implementing an event listener on the `FullPageCacheEvents::CACHE_RESPONSE` event) however, I think Pimcore should follow standards / best-practices here instead of requiring custom logic.

I'm happy to create a PR if you agree with this.

### PR DESCRIPTION
**Pull request by youwe-petervanderwal**
## Changes in this pull request  
Resolves #14429 

## Additional info  
Wasn't sure if this should go in the 10.6 or in the 11.x branch, let me know if I need to change it

### CODE REVIEW

File: doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md [hunk 1]
```diff
@@ -49,12 +49,43 @@ pimcore:
...
-In this case you can use obtain the pull page cache service from the container and disable it, eg. in an action: 
+### Disable caching via the response headers
+Adding the `Cache-Control: no-store` header to your response will disable the full page cache as well as any other
+middleware and browser caching:
+```php
+<?php
+$response->headers->addCacheControlDirective('no-store');
+```
+
+### Disable the full page cache via an event listener
```
Comment (youwe-petervanderwal) [inline doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md:64]: This feature already existed in the full page cache service, but wasn't documented yet, so I added that as welll

File: bundles/CoreBundle/src/EventListener/Frontend/FullPageCacheListener.php [hunk 2]
```diff
@@ -364,6 +364,11 @@ private function responseCanBeCached(Response $response): bool
     {
         $cache = true;
 
+        // do not cache when the application indicated a 'no-store' in the response Cache-Control header
+        if ($response->headers->getCacheControlDirective('no-store')) {
```
Comment (dvesh3) [inline bundles/CoreBundle/src/EventListener/Frontend/FullPageCacheListener.php]: Should we also respect "private" and "no-cache" headers here?

File: doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md [hunk 3]
```diff
@@ -49,12 +49,43 @@ pimcore:
 Sometimes it is more useful to deactivate the full page cache directly in the code, for example when 
 it's not possible to define an exclude-regex, or for similar reasons.
 
-In this case you can use obtain the pull page cache service from the container and disable it, eg. in an action: 
+### Disable caching via the response headers
+Adding the `Cache-Control: no-store` header to your response will disable the full page cache as well as any other
+middleware and browser caching:
+```php
+<?php
+$response->headers->addCacheControlDirective('no-store');
```
Comment (dvesh3) [inline doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md:61]: why not use the response helper example here? like this one https://github.com/pimcore/pimcore/blob/b9aa97ad9c1b7576feb17947de9022cd46a977b8/bundles/AdminBundle/src/Controller/Admin/LoginController.php#L84

**Comment by dvesh3** [comment]
@youwe-petervanderwal could you please resolve conflicts? and check my suggestion? thanks!

File: doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md (hunk 3 above)
Comment (youwe-petervanderwal) [reply doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md:61]: Happy to change it, but at the moment the ResponseHelper is marked as `@internal`. So then we should remove that flag. Please indicate if you want to go that route

https://github.com/pimcore/pimcore/blob/11.x/lib/Http/ResponseHelper.php

File: bundles/CoreBundle/src/EventListener/Frontend/FullPageCacheListener.php (hunk 2 above)
Comment (youwe-petervanderwal) [reply bundles/CoreBundle/src/EventListener/Frontend/FullPageCacheListener.php]: Good one, added those as well

File: doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md (hunk 3 above)
Comment (dvesh3) [reply doc/19_Development_Tools_and_Details/09_Cache/03_Full_Page_Cache.md:61]: yes, right. I didn't see that internal tag. Then it is fine like this

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@youwe-petervanderwal nice work 👍 thanks!

**Pull request merged**